png_benchmark.py

```diff
@@ -0,0 +1,152 @@
+as well as a .json with the test data and a .csv which provides a more human-friendly summary of the data.
+"""
+import gc
+import io
+import json
+import os
+import pathlib
+import time
+
+from PIL import Image
+
+
+# Setting parameters
+FRAME_COUNT = 10  # Total number of frames
+FRAME_TO_SAVE = 69  # This frame is good, it has both landing pads in it
+INPUT_PATH = pathlib.Path("test_images", "Encode Test Dataset 2024")
+OUTPUT_PATH = pathlib.Path(f"log_{int(time.time())}")
+COMPRESS_TYPES = [0, 1, 2, 3, 4]  # There are 5 different compression algorithms, each are numbered
+INITIAL_COMPRESS_LEVEL = 1  # Compress level 0 is skipped because it is uncompressed
+MAX_COMPRESS_LEVEL = 6  # Although this maxes out at 9, it takes way too long (like 10s per image)
+
+# Keys for dictionary entries
+MAX_TIME_MS = "max_time_ms"
+MIN_TIME_MS = "min_time_ms"
+AVG_TIME_MS = "avg_time_ms"
+MAX_SIZE_B = "max_size_B"
+MIN_SIZE_B = "min_size_B"
+AVG_SIZE_B = "avg_size_B"
+MAX_RATIO_COMPRESSED_TO_ORIGINAL = "max_size_ratio_compressed_to_original_%"
+MIN_RATIO_COMPRESSED_TO_ORIGINAL = "min_size_ratio_compressed_to_original_%"
+AVG_RATIO_COMPRESSED_TO_ORIGINAL = "avg_size_ratio_compressed_to_original_%"
+FRAME_DATA = "frame_data"
+
+
+if __name__ == "__main__":
+    OUTPUT_PATH.mkdir(parents=True, exist_ok=True)
+
+    results = {
+        f"compress_type_{compress_type}": {
+            f"compress_level_{compress_level}": {
+                MAX_TIME_MS: 0,
+                MIN_TIME_MS: 0,
+                AVG_TIME_MS: 0,
+                MAX_SIZE_B: 0,
+                MIN_SIZE_B: 0,
+                AVG_SIZE_B: 0,
+                MAX_RATIO_COMPRESSED_TO_ORIGINAL: 0,  # % of original size
+                MIN_RATIO_COMPRESSED_TO_ORIGINAL: 0,
+                AVG_RATIO_COMPRESSED_TO_ORIGINAL: 0,
+                FRAME_DATA: []
+            } for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL + 1)
+        } for compress_type in COMPRESS_TYPES
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+
+    for compress_type in COMPRESS_TYPES:
+        print(f"-----------------COMPRESS TYPE {compress_type}--------------------")
+        for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL + 1):
+            max_time_ns = 0
+            min_time_ns = float("inf")
+            total_time_ns = 0
+            max_size_B = 0
+            min_size_B = float("inf")
+            total_size_B = 0
+            max_compression_ratio = 0
+            min_compression_ratio = float("inf")
+            total_compression_ratio = 0
+            for frame_index in range(FRAME_COUNT):
+                img = Image.open(pathlib.Path(INPUT_PATH, f"{frame_index}.png"))
+                buffer = io.BytesIO()
+
+                # Running encode
+                gc.disable()
+                start = time.time_ns()
+                img.save(buffer, format="PNG", compress_level=compress_level, compress_type=compress_type)
```

Comment:
Line too long, maximum is 100 characters. Use multiline formatting for function calls.